app/controllers/ips_controller.rb

```diff
@@ -7,6 +7,7 @@ def create
     default_location = current_organisation.locations.first
     @ip = default_location.ips.new(ip_params)
     if @ip.save
+      publish_locations_and_ips_to_s3
```

Comment:
I'd remove the implementation-specific part of this naming - the `to_s3` part.

This controller doesn't need visibility of how the publishing is working (and it can be discovered by a reader from how the use case is composed with `Gateways::S3`).

Reply:
Perhaps `publish_locations_and_ips_for_performance_platform`? I think I reader should know what this line does without having to look much further.

Reply:
I like that, though I did overhear someone the other day speculating about other clients for this data. Happy for you to decide, naming is hard.